Source/DafnyDriver/Commands/VerifyCommand.cs

```diff
@@ -28,24 +27,22 @@ public static class VerifyCommand {
     foreach (var option in Options) {
       result.AddOption(option);
     }
-    DafnyNewCli.SetHandlerUsingDafnyOptionsContinuation(result, async (options, _) => {
-      if (options.Get(CommonOptionBag.VerificationCoverageReport) != null) {
-        options.TrackVerificationCoverage = true;
-      }
-
-      if (options.Get(CommonOptionBag.VerificationLogFormat).Any() || options.Get(CommonOptionBag.VerificationCoverageReport) != null) {
```

Comment:
Note this `if` is gone in `HandleVerification`, so `dafny verify` no longer switches to using `SynchronousCliCompilation` when one of those options is used, which is the main contribution of this PR.